pkg/core/resources/model/resource.go

```diff
@@ -718,6 +719,19 @@ func TargetRefToResourceIdentifier(meta ResourceMeta, tr common_api.TargetRef) R
 	}
 }
 
+func ResourceToBackendRef(r Resource, resType ResourceType, port uint32) common_api.BackendRef {
+	id := NewResourceIdentifier(r)
+	return common_api.BackendRef{
+		TargetRef: common_api.TargetRef{
+			Kind:      common_api.TargetRefKind(resType),
+			Name:      id.Name,
+			Namespace: id.Namespace,
+			Mesh:      id.Mesh,
```

Comment:
`Mesh` is reserved for cross-mesh, I doubt we actually take into account this value